### PR TITLE
Sort export records in reverse chronological order AB#11675

### PR DESCRIPTION
**Pull request by MikeLyttle**
# Implements [AB#11765](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/11765)

-   [x] Enhancement
-   [x] Bug
-   [ ] Documentation

## Description

- Swaps the sorting order from oldest-first to newest-first for all exports
- Adds newest-first sorting for immunization recommendations
- Fixes a bug where an empty table was sometimes displayed instead of the "No recommendations found." message

If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

## Testing

-   [ ] Unit Tests Updated
-   [ ] Functional Tests Updated
-   [ ] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

NO

### Browsers Tested

-   [x] Chrome
-   [ ] Safari
-   [ ] Edge
-   [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant. Include any specialized deployment steps here.

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

-   Fulfills Work Item Requirements

    -   The changes meet/implement story's acceptance criteria. Fixes identified problems if bug.
    -   Changes not directly related to the task requirements need to be documented on the PR.

-   Compilation / Tests

    -   The changes do not break compilation and/or unit or functional tests; unless other item addresses them specifically.

-   Logic Problems / Functional Behaviour

    -   The changes work as intended and do not introduce problems.

-   Performance

    -   The changes do not introduce obvious performance issues.

-   Documented Standards

    -   The changes adhere to the team's documented [Coding Standards](https://github.com/bcgov/healthgateway/wiki/standards).

-   Readability / Maintainability
    -   The changes can be easily understood/read and allow for future enhancements without major refactoring. Readability is preferred over "clever code".
    -   Reasoning for changes needs to be clearly articulated. Disagreements should be arbitrated by a third developer.

### CODE REVIEW

**Review by pagustinqt20 — APPROVED** [review]

File: Apps/WebClient/src/ClientApp/src/components/report/covid19.vue [hunk 1]
```diff
@@ -58,9 +58,9 @@ export default class COVID19ReportComponent extends Vue {
                 b.labResults[0].collectedDateTime
             );
 
-            return firstDate.isAfter(secondDate)
+            return firstDate.isBefore(secondDate)
                 ? 1
-                : firstDate.isBefore(secondDate)
+                : firstDate.isAfter(secondDate)
                 ? -1
                 : 0;
```
Comment (BrianMaki) [inline Apps/WebClient/src/ClientApp/src/components/report/covid19.vue:65]: Nit: are we not dealing with sonar lint issues?

Screen Shot 2021-11-18 at 11 26 38 AM

File: Apps/WebClient/src/ClientApp/src/components/report/immunizationHistory.vue [hunk 2]
```diff
@@ -101,23 +101,49 @@ export default class ImmunizationHistoryReportComponent extends Vue {
...
+            }
+
+            const firstDate = new DateWrapper(a.diseaseDueDate);
+            const secondDate = new DateWrapper(b.diseaseDueDate);
+
+            return firstDate.isBefore(secondDate)
+                ? 1
+                : firstDate.isAfter(secondDate)
+                ? -1
+                : 0;
```
Comment (BrianMaki) [inline Apps/WebClient/src/ClientApp/src/components/report/immunizationHistory.vue:130]: Nit: sonar lint issue as above.

File: Apps/WebClient/src/ClientApp/src/components/report/medicationHistory.vue [hunk 3]
```diff
@@ -69,9 +69,9 @@ export default class MedicationHistoryReportComponent extends Vue {
             const firstDate = new DateWrapper(a.dispensedDate);
             const secondDate = new DateWrapper(b.dispensedDate);
 
-            return firstDate.isAfter(secondDate)
+            return firstDate.isBefore(secondDate)
                 ? 1
-                : firstDate.isBefore(secondDate)
+                : firstDate.isAfter(secondDate)
                 ? -1
                 : 0;
```
Comment (BrianMaki) [inline Apps/WebClient/src/ClientApp/src/components/report/medicationHistory.vue:76]: Nit: sonar lint issue as above.

File: Apps/WebClient/src/ClientApp/src/components/report/mspVisits.vue [hunk 4]
```diff
@@ -51,9 +51,9 @@ export default class MSPVisitsReportComponent extends Vue {
             const firstDate = new DateWrapper(a.encounterDate);
             const secondDate = new DateWrapper(b.encounterDate);
 
-            return firstDate.isAfter(secondDate)
+            return firstDate.isBefore(secondDate)
                 ? 1
-                : firstDate.isBefore(secondDate)
+                : firstDate.isAfter(secondDate)
                 ? -1
                 : 0;
```
Comment (BrianMaki) [inline Apps/WebClient/src/ClientApp/src/components/report/mspVisits.vue:58]: Nit: sonar lint issue as above.

**Review by BrianMaki — APPROVED** [review]
Nit: Are we not dealing

**Comment by MikeLyttle** [comment]
> Nit: Are we not dealing with sonar lint issues?

Good question... we're not expecting them to be checked for each PR, but if you feel they are reporting a worthwhile code smell, you can include that in your review.

It shouldn't hurt to refactor this to remove the nested ternary operators.

**Review by pagustinqt20 — APPROVED** [review]

**Review by BrianMaki — APPROVED** [review]

**Pull request merged**